lib/simplifyExpression/functionsSearch/nthRoot.js

```diff
@@ -197,46 +198,61 @@ function getFactorNodes(node) {
 // into groups of the root as a step
 // e.g. nthRoot(2 * 2 * 2, 2) -> nthRoot((2 * 2) * 2, 2)
 function groupTermsByRoot(node) {
-  const newNode = clone(node);
+  let newNode = clone(node);
   const radicandNode = getRadicandNode(node);
   const rootNode = getRootNode(node);
   const rootValue = parseFloat(rootNode.value);
 
   radicandNode.args.sort(sortNodes);
 
-  // We want to go through the sorted nodes, and try to find any groups of the
-  // same node that are the size of the root value
-  let children = [], hasGroups = false;
-  for (let i = 0; i < radicandNode.args.length;) {
-    let j = i;
-    const initialNode = radicandNode.args[i];
-    while (j < radicandNode.args.length && j - i < rootValue) {
-      const siblingNode = radicandNode.args[j];
-      if (!initialNode.equals(siblingNode)) {
-        break;
-      }
-      j++;
-    }
-    if (j - i === rootValue) {
-      hasGroups = true;
-      const groupedNode = Node.Creator.parenthesis(
-        Node.Creator.operator('*', radicandNode.args.slice(i, j)));
-      children.push(groupedNode);
-    }
-    else {
-      children = children.concat(radicandNode.args.slice(i, j));
-    }
-    i = j;
-  }
+  const terms = radicandNode.args.map(arg => print(arg));
+
+  // Collect groups when there are duplicate terms in the radicand
+  if ([...new Set(terms)].length !== terms.length) {
+    // dictionary storing the number of times a constant appears
+    // e.g. 2 * 2 * 2 => {'2': 3} , 2 appears 3 times
+    const count = {};
+
+    radicandNode.args.forEach(arg => {
+      arg = print(arg);
+      count[arg] ? count[arg]++ : count[arg] = 1;
+    });
+
+    const flatten = arr => arr.reduce(
+      (acc, val) => acc.concat(val), []
+    );
+
+    newNode = Node.Creator.nthRoot(
+      Node.Creator.operator(
+        '*',
+        flatten(Object.keys(count).map(key => {
+          let leftover = count[key];
+          const term = math.parse(key);
+          const times = rootValue;
+
+          const args = [];
+
+          while (leftover - times > 0) {
```

Comment:
This test case should remove the need for that:
`['nthRoot(4, 3)', 'nthRoot(4, 3)'],`